util/profile/profile.go

```diff
@@ -139,3 +146,25 @@ func (c *Collector) ParseGoroutines(reader io.Reader) ([][]types.Datum, error) {
 	}
 	return rows, nil
 }
+
+// getFuncMemUsage get function memory usage from heap profile
+func (c *Collector) getFuncMemUsage(name string) (int64, error) {
+	prof := pprof.Lookup("heap")
```

Comment:
How expensive is this function call? would it impact the performance of other goroutines on this tidb-server evidently?

Reply:
This function scrape the heap information in a single goroutine without any mutex/resource contention. And 791af28 is the same as c1bee54, we run tpcc test for 791af28 and result showed that it won't affect the query tps/qps.